---
title: "An introduction to the bambu package using NanoporeRNASeq data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{NanoporeRNASeq}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---



```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning=FALSE, message=FALSE,
  comment = "##>"
)
```

##Introduction
*[NanoporeRNASeq](https://github.com/GoekeLab/NanoporeRNASeq)* contains single chromosome RNA-Seq data from two common cell lines K562 and MCF7. Each of these cell line has three replicates, with 1 direct RNA sequencing data and 2 cDNA sequencing data. 

##Accessing NanoporeRNASeq data
###Load the NanoporeRNASeq package
```{r setup}
library("NanoporeRNASeq")
```

###Get the paths to the bam files
```{r}
data("bamFiles")
readlist <- system.file("extdata", bamFiles, package="NanoporeRNASeq")
```

###Get the annotation GRangesList
```{r}
data("annotationGrangesList_chr22_1_25409234")
```

##Getting coverage information with GenomicAlignments
###Load GenomicAlignments and GenomicRanges
```{r}
library(GenomicAlignments)
library(GenomicRanges)
library(ggplot2)
```
###Get coverage of a single bam file
```{r, fig.width = 8, fig.height = 6}
#get the path of the first bam file in the readlist
##replace list index to get path for another file 
bamfile_name <- bamFiles[1]
bamfile_path <- readlist[1]
#load bam file into a GAlignments object
ga <- readGAlignments(bamfile_path)
#get coverage of the alignment
ga_coverage <- coverage(ga)
#visualize the coverage of a specified range (the last 10kb here)
##change the IRanges(start,end) to your region of interest
range <- GRanges("22",IRanges(25309234,25409234))
ga_coverage[range]
ga_cov_df <-data.frame(Coordinate = 25309234:25409234,
                       Coverage = as.numeric(ga_coverage$`22`[ranges(range)]))
ggplot(ga_cov_df, aes(x=Coordinate, y=Coverage, group=1))+
  geom_line()+
  labs(title=paste("Coverage Plot of", bamfile_name, sep=" "))
```

## Running Bambu with NanoporeRNASeq data
###Load the bambu package
```{r}
library(bambu)
library(BSgenome.Hsapiens.NCBI.GRCh38)
#packages for plotting
library(circlize)
library(ComplexHeatmap)
library(ggbio)
```

###Run bambu
```{r}
seExtended <- bambu(reads = readlist, annotations = annotationGrangesList_chr22_1_25409234, genomeSequence = "BSgenome.Hsapiens.NCBI.GRCh38", extendAnnotations = TRUE, verbose = FALSE, ncore = 1)
seExtended
##write bambu output to an output directory
writeBambuOutput(seExtended,"output/")
```
###Check and visualize the estimated transcript expression
Option 1: using heatmap:
```{r, fig.width = 8, fig.height = 6}
colData(seExtended)$groupVar <-  unlist(lapply(colnames(seExtended),function(x) unlist(strsplit(x,"_"))[2]))
colnames(seExtended) <- gsub("_genome_chr22_1_25409234","",colnames(seExtended))
colData(seExtended)$name <- gsub("_genome_chr22_1_25409234","",colData(seExtended)$name)
plot.bambu(seExtended, group.variable = "groupVar", type = "heatmap")
```
Option 2: using PCA plot
```{r, fig.width = 8, fig.height = 6}
colData(seExtended)$groupVar <-  unlist(lapply(colnames(seExtended),function(x) unlist(strsplit(x,"_"))[2]))
plot.bambu(seExtended, group.variable = "groupVar", type = "pca")
```

###Check for single gene examples
Single gene examples can also be checked using plot functions from *bambu*
```{r, fig.width = 8, fig.height = 10}
plot.bambu(seExtended, type = "annotation", gene_id = unique(rowData(seExtended)$GENEID)[10])
```

###Obtain gene expression from transcript expression
```{r}
seGene <- transcriptToGeneExpression(seExtended)
seGene
```

###Check and visualize gene expression 
Option 1: using heatmap
```{r, fig.width = 8, fig.height = 6}
colData(seGene)$groupVar <-  unlist(lapply(colnames(seGene),function(x) unlist(strsplit(x,"_"))[2]))
plot.bambu(seGene, group.variable = "groupVar", type = "heatmap")
```
Option 2: using PCA plot
```{r, fig.width = 8, fig.height = 6}
colData(seGene)$groupVar <-  unlist(lapply(colnames(seGene),function(x) unlist(strsplit(x,"_"))[2]))
plot.bambu(seGene, group.variable = "groupVar", type = "pca")
```
