---
title: "An introduction to the bambu package using NanoporeRNASeq data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{NanoporeRNASeq}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---



```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning=FALSE, message=FALSE,
  comment = "##>"
)
```

## Introduction
*[NanoporeRNASeq](https://github.com/GoekeLab/NanoporeRNASeq)* contains single chromosome RNA-Seq data from two common cell lines K562 and MCF7. Each of these cell line has three replicates, with 1 direct RNA sequencing data and 2 cDNA sequencing data. 

## Accessing NanoporeRNASeq data
### Load the NanoporeRNASeq package
```{r setup}
library("NanoporeRNASeq")
```

### Get the paths to the bam files
```{r}
data("bamFiles")
readlist <- system.file("extdata", bamFiles, package="NanoporeRNASeq")
```

### Get the annotation GRangesList
```{r}
data("annotation")
```

## Get coverage of a single bam file
coverage of a single gene ENST00000215832 (MAPK1)
```{r, fig.width = 12}
library(ggbio)
bamfile <- BamFile(readlist[1])
range <- annotation$ENST00000215832
autoplot(bamfile, which=range)
```
show mismatch proportion in ENST00000215832 (MAPK1)
```{r, fig.width = 12}
library(BSgenome.Hsapiens.NCBI.GRCh38)
autoplot(bamfile, stat="mismatch", which=range, bsgenome=Hsapiens)
```

## Running Bambu with NanoporeRNASeq data
### Load the bambu package
```{r}
library(bambu)
library(BSgenome.Hsapiens.NCBI.GRCh38)
#packages for plotting
library(circlize)
library(ComplexHeatmap)
library(ggbio)
```

### Run bambu
```{r}
seExtended <- bambu(reads = readlist, annotations = annotation, genomeSequence = "BSgenome.Hsapiens.NCBI.GRCh38", extendAnnotations = TRUE, verbose = FALSE, ncore = 1)
seExtended
```

### Check for single gene examples
Single gene examples can also be checked using plot functions from *bambu*
```{r, fig.width = 8, fig.height = 10}
plot.bambu(seExtended, type = "annotation", gene_id = unique(rowData(seExtended)$GENEID)[10])
```
